File: backend_Myecoprint/backend_myecoprint/models/reason_barcode.js

const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const reasonBarcode = sequelize.define('reasonBarcode', {
  reasonBarcode_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    primaryKey: true,
    autoIncrement: true,
  },
  reason_arabic: {
    type: DataTypes.STRING,
    allowNull: false,
  },

  reason_english: {
    type: DataTypes.STRING,
  },
  createdAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
  updatedAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
  deletedAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
}, {
    timestamps: false, 
    paranoid: true, 
    tableName: 'reason_barcodes', 
  });

module.exports = reasonBarcode;

File: My_ecoprint_app/my_eco_print/lib/data/module/reason_barcode.dart
class ReasonBarcode {
  int reasonBarcodeId;
  String reasonArabic;
  String? reasonEnglish;
  DateTime? createdAt;
  DateTime? updatedAt;
  DateTime? deletedAt;

  ReasonBarcode({
    required this.reasonBarcodeId,
    required this.reasonArabic,
    this.reasonEnglish,
  
  });

  factory ReasonBarcode.fromJson(Map<String, dynamic> json) {
    return ReasonBarcode(
      reasonBarcodeId: json['reasonBarcode_id']as int? ??0,
      reasonArabic: json['reason_arabic']as String? ?? '',
      reasonEnglish: json['reason_english']as String? ?? '',
    
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'reasonBarcode_id': reasonBarcodeId,
      'reason_arabic': reasonArabic,
      'reason_english': reasonEnglish,
   
    };
  }
}
